src/Microsoft.ML.ImageAnalytics/ImageStaticPipe.cs

```diff
@@ -0,0 +1,169 @@
+        /// </summary>
+        /// <param name="path">The scalar text column that holds paths to the images</param>
+        /// <param name="relativeTo">If specified, paths are considered to be relative to this directory.
+        /// However, since hte transform can be persisted across machines, it is generally considered more
+        /// safe for practitioners to simply always make their input path data contain absolute paths.</param>
+        /// <returns>The column shcema definition for </returns>
+        /// <seealso cref="ImageLoaderEstimator"/>
+        public static Scalar<UnknownSizeBitmap> LoadAsImage(this Scalar<string> path, string relativeTo = null)
+        {
+            Contracts.CheckValue(path, nameof(path));
+            Contracts.CheckValueOrNull(relativeTo);
+            return new ImageLoaderEstimator.OutPipelineColumn(path, relativeTo);
+        }
+
+        /// <summary>
+        /// Converts the image to grayscale.
+        /// </summary>
+        /// <param name="input">The image to convert</param>
+        /// <returns>The grayscale images</returns>
+        /// <seealso cref="ImageGrayscaleEstimator"/>
+        public static Scalar<UnknownSizeBitmap> AsGrayscale(this Scalar<UnknownSizeBitmap> input)
+        {
+            Contracts.CheckValue(input, nameof(input));
+            return new ImageGrayscaleEstimator.OutPipelineColumn<UnknownSizeBitmap>(input);
+        }
+
+        /// <summary>
+        /// Converts the image to grayscale.
+        /// </summary>
+        /// <param name="input">The image to convert</param>
+        /// <returns>The grayscale images</returns>
+        /// <seealso cref="ImageGrayscaleEstimator"/>
+        public static Scalar<Bitmap> AsGrayscale(this Scalar<Bitmap> input)
+        {
+            Contracts.CheckValue(input, nameof(input));
+            return new ImageGrayscaleEstimator.OutPipelineColumn<Bitmap>(input);
+        }
+
+        /// <summary>
+        /// Given a column of images of unfixed size, resize the images so they have uniform size.
+        /// </summary>
+        /// <param name="input">The input images</param>
+        /// <param name="width">The width to resize to</param>
+        /// <param name="height">The height to resize to</param>
+        /// <param name="resizing">The type of resizing to do</param>
+        /// <param name="cropAnchor">If cropping is necessary, at what </param>
+        /// <returns>The now uniformly sized images</returns>
+        /// <seealso cref="ImageResizerEstimator"/>
+        public static Scalar<Bitmap> Resize(this Scalar<UnknownSizeBitmap> input, int width, int height,
+            ImageResizerTransform.ResizingKind resizing = ImageResizerTransform.ResizingKind.IsoCrop,
+            ImageResizerTransform.Anchor cropAnchor = ImageResizerTransform.Anchor.Center)
+        {
+            Contracts.CheckValue(input, nameof(input));
+            Contracts.CheckParam(width > 0, nameof(width), "Must be positive");
```

Comment:
>CheckParam(width > 0, nameof(width), "Must be positive") [](start = 22, length = 56)

not for this PR, but maybe we can add a few more utility methods, like "CheckPositiveParam"

Reply:
Yes, we ought to change Contracts a bit... one of its major purposes is to standardize how exceptions are typed and thrown, but it doesn't always help with what the error messages actually look like, which seems like the most potentially useful thing it could do.

---
In reply to: [214718439](https://github.com/dotnet/machinelearning/pull/778#discussion_r214718439) [](ancestors = 214718439)